stubs/toml/toml/decoder.pyi

```diff
@@ -0,0 +1,79 @@
+import sys
+from _typeshed import SupportsRead
+from typing import Any, Callable, Generic, MutableMapping, Pattern, Protocol, Text, TypeVar, overload
+
+_MutableMappingT = TypeVar("_MutableMappingT", bound=MutableMapping[str, Any])
+_KT_contra = TypeVar("_KT_contra", contravariant=True)
+_VT = TypeVar("_VT")
+
+if sys.version_info >= (3, 0):
+    from pathlib import PurePath
+
+    FNFError = FileNotFoundError
+    _PathLike = str | bytes | PurePath
+else:
+    FNFError = IOError
+    _PathLike = str | bytes
+
+class _SupportsGetSetItem(Protocol[_KT_contra, _VT]):
+    def __getitem__(self, __k: _KT_contra) -> _VT: ...
+    def __setitem__(self, __k: _KT_contra, __v: _VT) -> None: ...
+
+TIME_RE: Pattern[str]
+
+class TomlDecodeError(ValueError):
+    msg: str
+    doc: str
+    pos: int
+    lineno: int
+    colno: int
+    def __init__(self, msg: str, doc: str, pos: int) -> None: ...
+
+class CommentValue(_SupportsGetSetItem[_KT_contra, _VT]):
+    val: _SupportsGetSetItem[_KT_contra, _VT]
+    comment: str
+    def __init__(
+        self, val: _SupportsGetSetItem[_KT_contra, _VT], comment: str, beginline: bool, _dict: type[_MutableMappingT]
+    ) -> None: ...
+    def __getitem__(self, key: _KT_contra) -> _VT: ...
+    def __setitem__(self, key: _KT_contra, value: _VT) -> None: ...
+    def dump(self, dump_value_func: Callable[[Any], str]) -> str: ...
+
+@overload
+def load(
+    f: _PathLike | list[_PathLike] | SupportsRead[Text],
```

Comment:
`list[_PathLike]` doesn't work. Unlike you would expect, it won't accept a list of strings: `list[str]` is not compatible with `list[str | PurePath]`. Look up invariance (or ask me to explain) if you want to learn why that is. It's explained pretty well in mypy's docs, for example.

I'd suggest using `Sequence[_PathLike]` or `list[Any]` instead, and add a comment that `list[_PathLike]` won't work because of invariance.

Reply:
I don't think `Sequence` would work here because of this line:
https://github.com/uiri/toml/blob/master/toml/decoder.py#L13523